### PR TITLE
[12.0][ADD] spp_custom: track visibility

### PR DESCRIPTION
**Pull request by mclaeysb**
Track visibility for a bunch of fields, including custom track visibility for M2M and O2M fields.

Not the most beautiful code I've written, but I don't know how to make it cleaner. Tips welcome.

FYI: I'm using `rec.taxes_id` before and after the write, instead of `vals[taxes_id]`, because the latter has complicated `[(6, _, [ids])]` values etc.

### CODE REVIEW

File: spp_custom/models/product.py [hunk 1]
```diff
@@ -13,6 +13,69 @@
...
+    uom_po_id = fields.Many2one(track_visibility="onchange")
+    default_code = fields.Char(track_visibility="onchange")
+    sale_ok = fields.Boolean(track_visibility="onchange")
+    available_in_pos = fields.Boolean(track_visibility="onchange")
+
+    @api.multi
+    def write(self, vals):
+        # Custom track visibility for
+        # taxes_id (M2M), seller_ids (M2M, supplier_taxes_id (O2M)
+        for rec in self:
```
Comment (robinkeunen) [inline spp_custom/models/product.py:63]: maybe use `for template in self`?

**Review by robinkeunen — CHANGES_REQUESTED** [review]
I think this is the better way to do it :-)

Only problem I see is that you return res from inside the loop although write can only be used on a single record (90% sure on this, not 100%).

```python
@api.multi
def write(self, vals):
    # Custom track visibility for
    # taxes_id (M2M), seller_ids (M2M, supplier_taxes_id (O2M)
    for rec in self:
        ...
        res = super(ProductTemplate, self).write(vals).   # called on self not rec !
        ...
    return res
```

could be


```python

@api.multi
def write(self, vals):
    # Custom track visibility for
    # taxes_id (M2M), seller_ids (M2M, supplier_taxes_id (O2M)
    ...
    res = super(ProductTemplate, product).write(vals)
    ...
    return res
```

**Comment by mclaeysb** [comment]
> I think this is the better way to do it :-)
> 
> Only problem I see is that you return res from inside the loop although write can only be used on a single record (90% sure on this, not 100%).
> 
> ```python
> @api.multi
> def write(self, vals):
>     # Custom track visibility for
>     # taxes_id (M2M), seller_ids (M2M, supplier_taxes_id (O2M)
>     for rec in self:
>         ...
>         res = super(ProductTemplate, self).write(vals).   # called on self not rec !
>         ...
>     return res
> ```
> 
> could be
> 
> ```python
> @api.multi
> def write(self, vals):
>     # Custom track visibility for
>     # taxes_id (M2M), seller_ids (M2M, supplier_taxes_id (O2M)
>     ...
>     res = super(ProductTemplate, product).write(vals)
>     ...
>     return res
> ```

This is a good point and I don't know how to solve this yet...

**Comment by robinkeunen** [comment]
We're in a custom module, if that passes functional tests, it's ok :-)

(but still, I'd call the super in the loop on the product rather than on the self.

**Review by robinkeunen — APPROVED** [review]

**Comment by mclaeysb** [comment]
With improvements as suggested by @remytms - *praised be his tallent*.

**Review by robinkeunen — APPROVED** [review]

**Comment by robinkeunen** [comment]
nice

**Pull request merged**